.github/workflows/check-pr.yml

```diff
@@ -84,3 +84,15 @@ jobs:
           echo "Pull request is labeled as 'do not merge'"
           echo "This workflow fails so that the pull request cannot be merged"
           exit 1
+
+  check-semver:
+    name: Check semver
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```

Comment:
Isn't the default of checkout to use the head? Could this be removed with the same outcome → simpler config?

Reply:
Ok, getting back to this change - I'd suggest documenting what's happening here a bit better. It's difficult to assume knowledge of what each property here represents for someone reading this, and so a comment would help explain both what this is doing and why it's necessary here.

Reply:
Added in fdad2c9e16bcc5f473c7e0d13f34079a8fb99ebc

Reply:
It's still "what is happening here"… Can we simplify that in some way? Maybe by using its own workflow file and `on: pull_request` or something like that which does exactly what we need already requiring less confusing configuration? Not sure whether that's possible?